backend/src/scraper/bestbuy_scraper.py

```diff
@@ -0,0 +1,54 @@
+from os import environ
+from typing import Dict
+import requests
+
+class BestBuyProduct:
+    @staticmethod
+    def _product_specs(product_id: str) -> Dict:
+        params = {
+            "product_id": product_id, # TODO: Figure out what to pass to API.
+            "api_key": environ.get("BESTBUY_API_KEY"), # TODO: Update API Key.
+        }
+        search = requests.get(f"https://api.bestbuy.com/v1/products(sku={params.get('product_id')})?apiKey={params.get('api_key')}&show=customerReviewAverage,customerReviewCount,details.name,details.value,image,manufacturer,modelNumber,name,regularPrice,sku,url&pageSize=10&format=json")
+        results = search.json()
+
+        if "error" in results or results.get("total") == 0:
+            return {
+                "basic_info": {
+                    "product_id": params.get("product_id"),
+                    "error": "The product id does not exist.",
+                }
+            }
+        else:
+            product_result = results.get("products", {})[0]
+            spec_highlights = product_result.get("details", [])
+            return {
+                "basic_info": {
+                    "us_item_id": product_result.get("us_item_id", ""),
+                    "product_id": product_result.get("sku"),
+                    "model_number": product_result.get("modelNumber"),
+                    "title": product_result.get("name"),
+                    "product_page_url": product_result.get("url"),
+                    "price": {
+                        "amount": product_result.get("regularPrice"),
+                        "currency": "CAD" if product_result.get("regularPrice") is not None else None,
```

Comment:
Does this mean that if the `regularPrice` currency is say "USD" or "Euro", we overwrite it to "CAD"?

Reply:
Yes but the API doesn't provide currency information so we're operating under the assumption everything is in USD (not CAD - my mistake) since the information is coming from bestbuy.com.